cdbot/cogs/cyber.py

```diff
@@ -155,6 +157,147 @@ def __init__(self, bot: Bot):
 
         await ctx.send(embed=embed)
 
+    @command()
+    async def readme(self, ctx: Context, operand: str = "", channel_id: int = 0, msg_send_interval: int = 0):
+        """
+        Allows generating, sending and manipulation of JSON file containing the info needed
+        to create and send the embeds for the #readme channel. Only ROOT_ROLE_ID users have
+        the permissions need to use this command.
+        """
+
+        README_SEND_ALIASES = ["create", "push", "generate", "send", "make", "build", "upload"]
+        README_RECV_ALIASES = ["fetch", "get", "pull", "download", "retrieve", "dm", "dl"]
+
+        operand = operand.lower()
+
+        # The supplied operand is incorrect.
+        if not (operand in README_SEND_ALIASES + README_RECV_ALIASES):
+            incorrect_operand_embed = Embed(
+                colour=0x673ab7,
+                description=":shrug: **Invalid readme operand supplied.**"
+            )
+            await ctx.message.delete()
+            await ctx.send(embed=incorrect_operand_embed)
+
+        # User missed out the channel_id for certain commands.
+        elif (channel_id == 0 and operand in README_SEND_ALIASES):
+            misssing_channel_embed = Embed(
+                colour=0xff5722,
+                description=":facepalm: **Whoops, you missed out the channel ID! Try again.**"
+            )
+            await ctx.message.delete()
+            await ctx.send(embed=misssing_channel_embed)
+
+        # Process the request.
+        elif ROOT_ROLE_ID in [msg_author.id for msg_author in ctx.message.author.roles]:
```

Comment:
Instead of having an if-statement to check for the Root role, a command [check](https://discordpy.readthedocs.io/en/rewrite/ext/commands/api.html#discord.ext.commands.has_role) would be a better method of checking, as it hides the function when users type in `:help`

Reply:
I absolutely agree with you there, it would be a better form of control. I've just deleted my testing server environment, so it'll probably have to be sometime tomorrow or day after, for me to make this change, unless someone, could re-factor the conditional. Whichever is sooner :+1:

Reply:
@fractalpenguin I've refactored the conditional, squashed a bug and added the decorator. Does this look good? :+1: